### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by boazsegev** [1byte2bytes/LotteHTTP#1]
Hi,

I was browsing Github and came across your cool project. I was thrilled to notice you're using facil.io.

I also noticed that there's a possible memory leak in this line:

https://github.com/1byte2bytes/LotteHTTP/blob/f99a273222be7d785a81fbb19de3c1bb048a22a8/main.c#L113-L117

The [facil.io `http_set_header` function](https://facil.io/0.7.x/http#http_set_header) takes memory "ownership" of a header's value, but **not** its name.

This approach to memory ownership allows for the use of the `HTTP_HEADER_CONTENT_TYPE` in this line: https://github.com/1byte2bytes/LotteHTTP/blob/f99a273222be7d785a81fbb19de3c1bb048a22a8/main.c#L111

It's also consistent with the memory "ownership" model for FIOBJ hash tables (where only value ownership is transferred when setting a key-value pair, but keys ownership isn't transferred).

In other words, either create a constant:

```c
// global static variable
static FIOBJ HTTP_HEADER_SERVER;
static FIOBJ HTTP_HEADER_SERVER_VALUE;
// ... initialize variable in `main()`
HTTP_HEADER_SERVER = fiobj_str_new("server", strsizeof("server"));
HTTP_HEADER_SERVER_VALUE = fiobj_str_new("LotteHTTP/0.9.0", strsizeof("LotteHTTP/0.9.0"));
// ...use in function:
http_set_header(
        request, 
        HTTP_HEADER_SERVER, 
        fiobj_dup(HTTP_HEADER_SERVER_VALUE)
    );
```

Or use a temporary string:

```c
FIOBJ header_tmp = fiobj_str_new("server", strsizeof("server"));
http_set_header(
        request, 
        header_tmp, 
        fiobj_str_new("LotteHTTP/0.9.0", strsizeof("LotteHTTP/0.9.0"))
    );
fiobj_free(header_tmp);
```

Good luck with the project!

**Comment by 1byte2bytes** [comment]
Thank you! I had noticed a small memory leak, but haven't put the time into tracking it down. Hopefully this helps!

I had made this as a small project to make my own web server software, I like to make my software (and websites) small and light to run on weird/weak hardware (me and a friend once got Linux and my Discord bot on a Lego Mindstorms brick!). I thought I should try getting Facil.io and Lua to work, since Facil seems like a fairly light web framework and Lua a light programming language.

As you may notice, C isn't my primary programming language, haha! Thanks for the help 😄
